py_vsys/utils/crypto/curve_25519.py

```diff
@@ -18,6 +18,7 @@ def gen_pri_key(rand32: bytes) -> bytes:
     Returns:
         bytes: The generated private key
     """
+    rand32 = bytes.fromhex((rand32.hex() + '.')[:-1])
```

Comment:
This solution looks good.
Guess we can remove the `.` so that we have something like
```
rand32 = bytes.fromhex(rand32.hex())
```